### ISSUE TITLE
Auto Synchronize fail on postgresql composite key

### ISSUE THREAD

**Issue by RDeluxe** [typeorm/typeorm#1311]
Hi,

I have a composite key composed of 2 foreign key : 

``` javascript
import { GreenFrameState } from 'entities/greenFrameState.entity';
import { BrickTemplate } from 'entities/brickTemplate.entity';
import { Entity, Column, BaseEntity, ManyToOne } from 'typeorm';

@Entity()
export class BrickTemplatesGreenFrameStates extends BaseEntity {
  @ManyToOne(brickTemplate => BrickTemplate, brickTemplate => brickTemplate.brickTemplatesGreenFrameStates, { primary: true })
  public brickTemplate: BrickTemplate;

  @ManyToOne(greenFrameState => GreenFrameState, greenFrameState => greenFrameState.brickTemplatesGreenFrameStates, { primary: true })
  public greenFrameState: GreenFrameState;

  @Column()
  public step: number;
}
```

The first database creation works fine, but afterward I'm getting this error : 

```
executing query:  ALTER TABLE "public"."brick_templates_green_frame_states" DROP CONSTRAINT "fk_42e90aa0e824d5a53b909a0f429"
executing query:  ALTER TABLE "public"."brick_templates_green_frame_states" DROP CONSTRAINT "fk_8137afa857960ed14a49c07e268"
executing query:  ALTER TABLE "public"."brick_templates_green_frame_states" ALTER COLUMN "brickTemplateId" DROP NOT NULL
executing query:  ALTER TABLE "public"."brick_templates_green_frame_states" ALTER COLUMN "greenFrameStateId" DROP NOT NULL
query failed:  ALTER TABLE "public"."brick_templates_green_frame_states" ALTER COLUMN "brickTemplateId" DROP NOT NULL
error:  { error: column "brickTemplateId" is in a primary key
    at Connection.parseE (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:546:11)
    at Connection.parseMessage (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:371:19)
    at Socket.<anonymous> (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:114:22)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:252:12)
    at readableAddChunk (_stream_readable.js:239:11)
    at Socket.Readable.push (_stream_readable.js:197:10)
    at TCP.onread (net.js:589:20)
  name: 'error',
  length: 110,
  severity: 'ERROR',
  code: '42P16',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'tablecmds.c',
  line: '5243',
  routine: 'ATExecDropNotNull' }
executing query:  ROLLBACK
query failed:  ALTER TABLE "public"."brick_templates_green_frame_states" ALTER COLUMN "greenFrameStateId" DROP NOT NULL
error:  { error: current transaction is aborted, commands ignored until end of transaction block
    at Connection.parseE (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:546:11)
    at Connection.parseMessage (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:371:19)
    at Socket.<anonymous> (C:\Users\Deluxe\Dropbox\Moi\Job\Projects\lonestone\amplify-nature-api\node_modules\pg\lib\connection.js:114:22)
    at emitOne (events.js:115:13)
    at Socket.emit (events.js:210:7)
    at addChunk (_stream_readable.js:252:12)
    at readableAddChunk (_stream_readable.js:239:11)
    at Socket.Readable.push (_stream_readable.js:197:10)
    at TCP.onread (net.js:589:20)
  name: 'error',
  length: 143,
  severity: 'ERROR',
  code: '25P02',
  detail: undefined,
  hint: undefined,
  position: undefined,
  internalPosition: undefined,
  internalQuery: undefined,
  where: undefined,
  schema: undefined,
  table: undefined,
  column: undefined,
  dataType: undefined,
  constraint: undefined,
  file: 'postgres.c',
  line: '994',
  routine: 'exec_simple_query' }
```

Version : 0.1.7

**Comment by RDeluxe** [comment]
This may be linked to the note here, line 158 ?

https://github.com/typeorm/typeorm/blob/master/src/schema-builder/RdbmsSchemaBuilder.ts#L158

**Comment by pleerock** [comment]
provide `BrickTemplate` entity code as well

**Comment by RDeluxe** [comment]
Sure thing.

``` javascript
import { Entity, Column, PrimaryGeneratedColumn, BaseEntity, OneToMany } from 'typeorm';
import { Brick, BrickTemplatesGreenFrameStates, MosAvailableAction } from 'entities';

@Entity()
export class BrickTemplate extends BaseEntity {
  @PrimaryGeneratedColumn('uuid')
  public id: string;

  @Column()
  public name: string;

  @Column('text')
  public description: string;

  @Column({ unique: true })
  public code: string;

  @Column({ default: false })
  public aquatic: boolean;

  @OneToMany(type => BrickTemplatesGreenFrameStates,
    brickTemplatesGreenFrameStates => brickTemplatesGreenFrameStates.brickTemplate)
  public brickTemplatesGreenFrameStates: BrickTemplatesGreenFrameStates[];

  @OneToMany(type => Brick, brick => brick.brickTemplate)
  public bricks: Brick[];

  @OneToMany(type => MosAvailableAction, mosAvailableAction => mosAvailableAction.sourceBrickTemplate)
  public mosAvailableActionsAsSource: MosAvailableAction[];

  @OneToMany(type => MosAvailableAction, mosAvailableAction => mosAvailableAction.targetBrickTemplate)
  public mosAvailableActionsAsTarget: MosAvailableAction[];
}
```

**Comment by pleerock** [comment]
Fixed and will be released in `0.1.8`.
